### ISSUE TITLE
MAYBE_UNUSED causes compiling failure

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#1407]
In the definition of `MAYBE_UNUSED`, if `__has_cpp_attribute` is defined but `__has_cpp_attribute(maybe_unused)` is not, then `MAYBE_UNUSED` is not defined and causes compililation failures.
https://github.com/boutproject/BOUT-dev/blob/13fda6d569486bfda4c100c8547391ade2a15ee8/include/unused.hxx#L43-L51

**Comment by ZedThree** [comment]
Oops, that should fall-through in a sensible way. Sorry!

Out of curiosity, what compiler did this fail with?

**Comment by johnomotani** [comment]
My compiler is
```
$ mpic++ --version
g++ (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by ZedThree** [comment]
Fixed in #1409
